### PR TITLE
refactor: simplification of Picker and SearchBar to match FormInput

### PR DESCRIPTION
**Pull request by vhu-axelor**
RM#70566

### CODE REVIEW

File: packages/ui/src/components/organisms/Picker/Picker.tsx [hunk 1]
```diff
@@ -118,99 +118,87 @@ const Picker = ({
...
     }
 
     return null;
-  }, [emptyValue, isScrollViewContainer, listItems, pickerIsOpen]);
+  }, [emptyValue, isScrollViewContainer, listItems, isOpen]);
+
+  const _readOnly = useMemo(() => disabled || readOnly, [disabled, readOnly]);
+
+  const _required = useMemo(
+    () => (_readOnly || required) && selectedItem === null,
```
Comment (lme-axelor) [inline packages/ui/src/components/organisms/Picker/Picker.tsx:136]: Why are you using readonly here ?

File: packages/ui/src/components/organisms/Picker/Picker.tsx [hunk 2]
```diff
@@ -118,99 +118,87 @@ const Picker = ({
...
+    [Colors, _required, marginBottom, isOpen],
   );
 
+  if (_readOnly) {
+    return (
+      <FormInput
+        style={[styles.container, style]}
+        title={title}
+        defaultValue={
+          disabledValue == null || disabledValue === '' ? '-' : disabledValue
```
Comment (lme-axelor) [inline packages/ui/src/components/organisms/Picker/Picker.tsx:156]: I think you can use checkNullString here

File: packages/ui/src/components/organisms/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -53,63 +60,107 @@ const SearchBar = ({
   disableSearchPress = false,
 }: SearchBarProps) => {
   const Colors = useThemeColor();
-  const commonStyles = useMemo(() => getCommonStyles(Colors), [Colors]);
+
+  const _required = useMemo(
+    () => required && (valueTxt === null || valueTxt === ''),
```
Comment (lme-axelor) [inline packages/ui/src/components/organisms/SearchBar/SearchBar.tsx:65]: you can use checkNullString for valueTxt

File: packages/ui/src/components/organisms/SearchBar/SearchBar.tsx [hunk 4]
```diff
@@ -53,63 +60,107 @@ const SearchBar = ({
...
+        onSelection={onSelection}
+        onEndFocus={onEndFocus}
+        rightIconsList={[
+          <Icon
+            style={styles.action}
+            name="times"
+            color={Colors.secondaryColor_dark.background}
+            size={20}
+            touchable={true}
+            visible={valueTxt != null && valueTxt !== ''}
```
Comment (lme-axelor) [inline packages/ui/src/components/organisms/SearchBar/SearchBar.tsx:113]: simplify

File: packages/apps/crm/src/screens/lead/LeadFormScreen.js [hunk 5]
```diff
@@ -227,12 +226,6 @@ const styles = StyleSheet.create({
...
   },
-  civilityTitle: {
-    marginLeft: -10,
-  },
-  pickerTitle: {
-    marginLeft: 5,
-  },
   input: {
     width: '90%',
   },
```
Comment (gca-axelor) [inline packages/apps/crm/src/screens/lead/LeadFormScreen.js:231]: not used

File: packages/apps/crm/src/screens/contact/ContactFormScreen.js [hunk 6]
```diff
@@ -191,9 +191,6 @@ const styles = StyleSheet.create({
   containerZIndex: {
     zIndex: 40,
   },
-  marginPicker: {
-    marginLeft: 5,
-  },
   picker: {
     width: '100%',
   },
```
Comment (gca-axelor) [inline packages/apps/crm/src/screens/contact/ContactFormScreen.js:196]: not used

File: packages/apps/crm/src/screens/opportunity/OpportunityFormScreen.js [hunk 7]
```diff
@@ -241,9 +242,6 @@ const styles = StyleSheet.create({
   marginTitle: {
     marginLeft: 28,
   },
-  marginPicker: {
-    marginLeft: 5,
-  },
   button_container: {
     marginVertical: '1%',
     flexDirection: 'column',
```
Comment (gca-axelor) [inline packages/apps/crm/src/screens/opportunity/OpportunityFormScreen.js:247]: not used

File: packages/apps/helpdesk/src/screens/TicketFormScreen.js [hunk 8]
```diff
@@ -223,8 +214,6 @@ const TicketFormScreen = ({navigation, route}) => {
             onChange={value =>
               handleTicketFieldChange(value, 'responsibleUser')
             }
-            style={styles.picker}
-            styleTxt={styles.marginTitle}
           />
```
Comment (gca-axelor) [inline packages/apps/helpdesk/src/screens/TicketFormScreen.js:217]: a lot of css not used in this screen picker,pickerTitle,marginTitle,search

File: packages/apps/crm/src/screens/lead/LeadFormScreen.js (hunk 5 above)
Comment (gca-axelor) [reply packages/apps/crm/src/screens/lead/LeadFormScreen.js:231]: button_container

**Pull request merged**